sql-plugin/src/main/java/com/nvidia/spark/rapids/GpuColumnVector.java

```diff
@@ -158,6 +158,15 @@ private static final DType toRapidsOrNull(DataType type) {
       return DType.TIMESTAMP_MICROSECONDS;
     } else if (type instanceof StringType) {
       return DType.STRING;
+    } else if (type instanceof DecimalType) {
+      DecimalType decType = (DecimalType) type;
+      if (decType.precision() <= DType.DECIMAL32_MAX_PRECISION) {
```

Comment:
Spark is kind of split on how decimals are stored internally.  For `UnsafeRow` only supports a `long` value or a binary representation for `BigDecimal`.

https://github.com/apache/spark/blob/0b557b329046c66ee67a8c94c5bb95ffbe50e135/sql/catalyst/src/main/java/org/apache/spark/sql/catalyst/expressions/UnsafeRow.java#L382-L395

For the build in `ColumnVector` implementation they distinguish between int and long backed storage.

https://github.com/apache/spark/blob/0b557b329046c66ee67a8c94c5bb95ffbe50e135/sql/core/src/main/java/org/apache/spark/sql/execution/vectorized/WritableColumnVector.java#L361-L375

If we only support DECIMAL64 then it would potentially reduce complexity by a lot. But if we also support DECIMAL32 we could save on GPU memory.

The main reason I ask is because I just merged in some code that speeds up the transfer of fixed width data types.  I have not tested it with decimal in any way, but there is code in there that makes some assumptions about what the size of the data types are and I think it is wrong right now.  Should be simple to fix it, but I wanted to call this out and be sure we are all on the same page.

Reply:
Maybe we can start from only supporting DECIMAL64 for reducing complexity?

Reply:
I am fine either way, and it is not a big deal. I was just thinking out loud.